transports/janus_mqtt.c

```diff
@@ -1224,7 +1228,16 @@ int janus_mqtt_client_disconnect(janus_mqtt_context *ctx) {
 
 	options.context = ctx;
 	options.timeout = ctx->disconnect.timeout;
-	return MQTTAsync_disconnect(ctx->client, &options);
+	int rc = MQTTAsync_disconnect(ctx->client, &options);
+	if (rc == MQTTASYNC_SUCCESS) {
+		janus_mutex_lock(&ctx->disconnect.mutex);
+		// Block the thread awaiting asynchronous graceful disconnect to finish.
```

Comment:
Code style: comments should be in `/* */` format.

Reply:
Fixed